js_modules/dagit/packages/core/src/plugins/ipynb.tsx

```diff
@@ -22,6 +22,20 @@ export const SidebarComponent: React.FC<IPluginSidebarProps> = (props) => {
     return <span />;
   }
 
+  const value = notebookPath.value;
+  const url = React.useMemo(() => {
+    try {
+      const url = new URL(value);
+      return url.toString();
+    } catch (e) {
+      // Not a full valid URL
+      return null;
+    }
+  }, [value]);
```

Comment:
Ah, yeah, the hook warning is valid.

Hooks cannot be called conditionally, which means this will need to go above the `!notebookPath` conditional. To accommodate the behavior of the hook, you can do:

```
const value = notebookPath?.value || '';
const url = React.useMemo(() => {
  // same behavior as before
}, [value]);

// This conditional has to be below the hook.
if (!notebookPath) {
  return <span />;
}

// Otherwise, do stuff with URL
```

Reply:
fixed!